presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/TeradataDateFunctions.java

```diff
@@ -35,23 +39,30 @@
 import static com.facebook.presto.common.type.TimeZoneKey.getTimeZoneKeys;
 import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
 import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
-import static com.facebook.presto.teradata.functions.dateformat.DateFormatParser.createDateTimeFormatter;
 import static com.google.common.base.Throwables.throwIfInstanceOf;
 import static io.airlift.slice.Slices.utf8Slice;
 import static java.nio.charset.StandardCharsets.UTF_8;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 public final class TeradataDateFunctions
 {
+    private static final ThreadLocalCache<Slice, DateTimeFormatter> DATETIME_PARSER_CACHE =
```

Comment:
lets add a comment telling why we need a different parser and formatter

Reply:
Resolved in 2b89c16